src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -1660,79 +1646,36 @@ void network::transfer_memory_to_device(std::shared_ptr<primitive_inst> instance
+    return _variables_states.find(variable_id) != _variables_states.end();
 }
 
-void network::update_variable_memory(const std::string& variable_id, const cldnn::layout& layout) {
+ov::intel_gpu::VariableState& network::get_variable(const std::string &variable_id) const {
     auto it = _variables_states.find(variable_id);
-    if (it == _variables_states.end()) {
-        cldnn::network::VariableState::Ptr variable_state = std::make_shared<cldnn::network::VariableState>(get_engine().allocate_memory(layout, false));
-        _variables_states.insert({variable_id, variable_state});
-    } else {
-        bool can_reuse = it->second->memory && layout.count() <= it->second->memory->get_layout().count();
-        if (can_reuse)
-            it->second->set_memory(get_engine().reinterpret_buffer(*it->second->memory, layout));
-        else
-            it->second->set_memory(get_engine().allocate_memory(layout, false));
-        it->second->is_set = false;
-    }
-    for (auto primitive : _variable_state_primitives) {
-        if (const auto& memory_state_primitive = std::dynamic_pointer_cast<memory_state::variable>(primitive)) {
-            if (!variable_id.compare(memory_state_primitive->variable_id())) {
-                auto& variable_state = get_variable_memory(variable_id);
-                primitive->set_output_memory(variable_state.memory, false);
-            }
-        }
-    }
+    OPENVINO_ASSERT(it != _variables_states.end(), "[GPU] ", variable_id, " variable not found");
+    return *it->second;
+}
+const ov::intel_gpu::VariableStateInfo& network::get_variable_info(const std::string &variable_id) const {
+    auto it = _variables_state_info.find(variable_id);
+    OPENVINO_ASSERT(it != _variables_state_info.end(), "[GPU] ", variable_id, " variable info not found");
+    return it->second;
 }
 
-void network::allocate_variables_memories() {
-    for (const auto& info : _variables_state_info) {
-        auto variable_layout = info.second;
-        if (variable_layout.is_static()) {
-            _variables_states.insert({info.first, std::make_shared<cldnn::network::VariableState>(get_engine().allocate_memory(variable_layout, false))});
-        }
-    }
+const ov::intel_gpu::VariablesMap& network::get_variables() const {
+    return _variables_states;
 }
 
-const cldnn::network::variables_state_info_map& network::get_variables_state_info() const {
+const ov::intel_gpu::VariablesInfoMap& network::get_variables_info() const {
     return _variables_state_info;
 }
 
-void network::set_variables_state_info(const std::string& variable_id, const cldnn::layout& layout) {
-    auto it = _variables_state_info.find(variable_id);
-    if (it == _variables_state_info.end()) {
-        _variables_state_info.insert({variable_id, layout});
-    }
+void network::set_variables_state_info(const std::string& variable_id, const layout& variable_layout, std::shared_ptr<primitive_inst> inst) {
```

Comment:
`shared_ptr` for primitive_inst doesn't seem to be used here

Reply:
Removed